test/Microsoft.DotNet.ShellShim.Tests/Microsoft.DotNet.ShellShim.Tests.csproj

```diff
@@ -9,6 +9,7 @@
   </PropertyGroup>
   <ItemGroup>
     <ProjectReference Include="..\..\src\Microsoft.DotNet.Configurer\Microsoft.DotNet.Configurer.csproj" />
+    <ProjectReference Include="..\dotnet.Tests\dotnet.Tests.csproj" />
```

Comment:
This seems weird. Why is one test depending on another?

Reply:
The end goal is, simulator need to run the same test as production code. Simulators are under dotnet.Tests, because the simulator are supposed to be used in "core" code's test.

And it should not be shared outside the "core code", since simulator is the behavior from "core"'s point of view.  That's why I don't want to put it into test utility.

Maybe a new test dll?

https://res.infoq.com/news/2014/10/ddd-onion-architecture/en/resources/onion-architecture.png

Reply:
Move the simulators out somewhere